pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -132,20 +138,26 @@ type reconciler struct {
 }
 
 func (rc *reconciler) Run(sourcesReady config.SourcesReady, stopCh <-chan struct{}) {
-	wait.Until(rc.reconciliationLoopFunc(sourcesReady), rc.loopSleepDuration, stopCh)
+	// Wait for the populator to indicate that it has actually populated the desired state of world, meaning it has
+	// completed a populate loop that started after sources are all ready. After, there's no need to keep checking.
+	wait.PollUntil(rc.loopSleepDuration, func() (bool, error) {
+		rc.reconciliationLoopFunc(sourcesReady, rc.populatorHasAddedPods())()
+		return rc.populatorHasAddedPods(), nil
+	}, stopCh)
+	wait.Until(rc.reconciliationLoopFunc(sourcesReady, true), rc.loopSleepDuration, stopCh)
```

Comment:
yes. No it's not technically necessary, I thought it would make the code clearer where/why populatroHasAddedPods is getting called since after discussion with gnufied it was not immediately clear. I can trim it back if you want, I could see how this looks like premature optimization :)